chanbackup/single.go

```diff
@@ -126,11 +129,21 @@ func NewSingle(channel *channeldb.OpenChannel,
 	// key.
 	_, shaChainPoint := btcec.PrivKeyFromBytes(btcec.S256(), b.Bytes())
 
+	// If a channel is unconfirmed, the block height of the ShortChannelID
+	// is zero. This will lead to problems when trying to restore that
+	// channel as the spend notifier would get a height hint of zero.
+	// To work around that problem, we add the channel broadcast height
+	// to the channel ID so we can use that as height hint on restore.
+	chanID := channel.ShortChanID()
+	if chanID.BlockHeight == 0 {
```

Comment:
Also occurs to me that this is needed for the case where a user has a _dated_ backup, and the channel has already confirmed, but was still pending at the time the backup was created. They should still be able to use this to close+restore the ultimate channel (for both the initiator and the responder).

Reply:
Yes, exactly.